lib/tests/formulae.rb

```diff
@@ -82,27 +82,25 @@ def detect_formulae!(args:)
             "#{@argument} is not detected from GitHub Actions or a formula name!"
         end
 
-        # Use GitHub Actions variables for pull request jobs.
-        if ENV["GITHUB_BASE_REF"].present? && ENV["GITHUB_SHA"].present?
-          diff_start_sha1 = rev_parse("origin/#{ENV["GITHUB_BASE_REF"]}")
-          if diff_start_sha1.blank?
-            test git, "-C", repository, "fetch", "--depth=1",
-                 "origin", "+refs/heads/#{ENV["GITHUB_BASE_REF"]}",
-                 args: args
+        if tap.present? && ENV["GITHUB_REPOSITORY"].present? && tap.full_name.casecmp(ENV["GITHUB_REPOSITORY"]).zero?
+          # Use GitHub Actions variables for pull request jobs.
+          if ENV["GITHUB_BASE_REF"].present?
             diff_start_sha1 = rev_parse("origin/#{ENV["GITHUB_BASE_REF"]}")
+            if diff_start_sha1.blank?
+              test git, "-C", repository, "fetch", "--depth=1",
+                   "origin", "+refs/heads/#{ENV["GITHUB_BASE_REF"]}",
+                   args: args
+              diff_start_sha1 = rev_parse("origin/#{ENV["GITHUB_BASE_REF"]}")
+            end
+            diff_end_sha1 = ENV["GITHUB_SHA"]
+          # Use GitHub Actions variables for branch jobs.
+          else
+            diff_end_sha1 = diff_start_sha1 = ENV["GITHUB_SHA"]
           end
-          diff_end_sha1 = ENV["GITHUB_SHA"]
-        # Use GitHub Actions variables for branch jobs.
-        elsif ENV["GITHUB_SHA"].present?
-          diff_end_sha1 = diff_start_sha1 = ENV["GITHUB_SHA"]
-        # Otherwise just use the current SHA-1 (which may be overriden later)
-        else
-          if ENV["CI"] || ENV["GITHUB_ACTIONS"]
-            onoe <<~EOS
-              No known CI provider detected! If you are using GitHub Actions then we cannot find the expected  environment variables! Check you have e.g. exported them to a Docker container.
-            EOS
-          end
-          diff_end_sha1 = diff_start_sha1 = current_sha1
+        elsif ENV["GITHUB_REPOSITORY"].blank? && (ENV["CI"] || ENV["GITHUB_ACTIONS"])
+          onoe <<~EOS
+            No known CI provider detected! If you are using GitHub Actions then we cannot find the expected environment variables! Check you have e.g. exported them to a Docker container.
+          EOS
```

Comment:
```suggestion
        elsif ENV["CI"]
          onoe <<~EOS
            No known CI provider detected! If you are using GitHub Actions then we cannot find the expected environment variables! Check you have e.g. exported them to a Docker container.
          EOS
        elsif ENV["GITHUB_ACTIONS"]
           odie <<~EOS
            We cannot find the needed GitHub Actions environment variables! Check you have e.g. exported them to a Docker container.
          EOS
        end
```

Reply:
Still need to make sure we don't do these if it's the tap comparison which failed, but I think I can just reorder the checks.